db/mongodb.go

```diff
@@ -0,0 +1,394 @@
+	Municipio                        *string          `json:"municipio"`
+	Telefone1                        string           `json:"ddd_telefone_1"`
+	Telefone2                        string           `json:"ddd_telefone_2"`
+	Fax                              string           `json:"ddd_fax"`
+	Email                            *string          `json:"email"`
+	SituacaoEspecial                 string           `json:"situacao_especial"`
+	DataSituacaoEspecial             *string          `json:"data_situacao_especial"`
+	OpcaoPeloSimples                 *bool            `json:"opcao_pelo_simples"`
+	DataOpcaoPeloSimples             *string          `json:"data_opcao_pelo_simples"`
+	DataExclusaoDoSimples            *string          `json:"data_exclusao_do_simples"`
+	OpcaoPeloMEI                     *bool            `json:"opcao_pelo_mei"`
+	DataOpcaoPeloMEI                 *string          `json:"data_opcao_pelo_mei"`
+	DataExclusaoDoMEI                *string          `json:"data_exclusao_do_mei"`
+	RazaoSocial                      string           `json:"razao_social"`
+	CodigoNaturezaJuridica           *int             `json:"codigo_natureza_juridica"`
+	NaturezaJuridica                 *string          `json:"natureza_juridica"`
+	QualificacaoDoResponsavel        *int             `json:"qualificacao_do_responsavel"`
+	CapitalSocial                    *float32         `json:"capital_social"`
+	CodigoPorte                      *int             `json:"codigo_porte"`
+	Porte                            *string          `json:"porte"`
+	EnteFederativoResponsavel        string           `json:"ente_federativo_responsavel"`
+	DescricaoPorte                   string           `json:"descricao_porte"`
+	QSA                              []QSA            `json:"qsa"`
+	CNAESecundarios                  []CNAESecundario `json:"cnaes_secundarios"`
+}
+
+type MongoDB struct {
+	Client   *mongo.Client
+	Database *mongo.Database
+	Context  context.Context
+}
+
+// NewMongoDB inicializa uma nova conexão MongoDB encapsulada em uma estrutura.
+func NewMongoDB(dbName string) (MongoDB, error) {
+	uri := os.Getenv("DATABASE_URL")
+
+	ctx, _ := context.WithCancel(context.Background())
+
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(uri))
+	if err != nil {
+		return MongoDB{}, fmt.Errorf("erro ao conectar ao MongoDB: %w", err)
+	}
+
+	// Verifica a conexão
+	if err := client.Ping(ctx, nil); err != nil {
+		return MongoDB{}, fmt.Errorf("erro ao pingar no MongoDB: %w", err)
+	}
+
+	return MongoDB{
+		Client:   client,
+		Database: client.Database(dbName),
+		Context:  ctx,
+	}, nil
+}
+
+// CreateCollection cria a coleção especificada, se ainda não existir.
+func (m *MongoDB) CreateCollection() error {
+
+	if err := m.Database.CreateCollection(m.Context, companyTableName); err != nil {
+		return fmt.Errorf("erro ao criar a coleção: %w", err)
+	}
+
+	fmt.Println("Coleção criada com sucesso:", companyTableName)
```

Comment:
* Não usamos o `fmt.Println`, mas o módulo de log
* As saídas são em inglês

```suggestion
	log.Output(1, fmt.Sprintf("Collection %s created successfully", companyTableName))
```

Podes ajustar isso no restante do arquivo tb?

Reply:
corrigido.